From http://xenbits.xen.org/hg/linux-2.6.18-xen.hg/rev/c3b5fe4e94c6
From: xen-devel@lists.xenproject.org
Patch-mainline: Never, SUSE-Xen specific
Subject: netback: don't use last request to determine minimum Tx credit
References: bsc#957988 CVE-2015-8550 XSA-155

The last from guest transmitted request gives no indication about the
minimum amount of credit that the guest might need to send a packet
since the last packet might have been a small one.

Instead allow for the worst case 128 KiB packet.

This is part of CVE-2015-8550 / XSA-155.

Signed-off-by: David Vrabel <david.vrabel@citrix.com>
Reviewed-by: Wei Liu <wei.liu2@citrix.com>
Committed-by: Jan Beulich <jbeulich@suse.com>
Acked-by: jbeulich@suse.com

--- a/drivers/xen/netback/netback.c
+++ b/drivers/xen/netback/netback.c
@@ -867,9 +867,7 @@ static void tx_add_credit(netif_t *netif
 	 * Allow a burst big enough to transmit a jumbo packet of up to 128kB.
 	 * Otherwise the interface can seize up due to insufficient credit.
 	 */
-	max_burst = RING_GET_REQUEST(&netif->tx, netif->tx.req_cons)->size;
-	max_burst = min(max_burst, 131072UL);
-	max_burst = max(max_burst, netif->credit_bytes);
+	max_burst = max(131072UL, netif->credit_bytes);
 
 	/* Take care that adding a new chunk of credit doesn't wrap to zero. */
 	max_credit = netif->remaining_credit + netif->credit_bytes;
